Improve build instructions for libcu++ (#2881)
NVIDIA/cccl: docs/libcudacxx/setup/building_and_testing.rst

Reviewer: suggestion: A lot of the stuff that is in here is redundant with what is in the top-level docs here: https://github.com/NVIDIA/cccl/blob/main/CONTRIBUTING.md#building-and-testing

I'd prefer to consolidate towards unified docs like these. 

I do recognize the need and value for having specific instructions for how to use `lit`.

I'd propose deleting this entire file and instead starting a "libcudacxx Developer Guide" document similar to the CUB one's here: https://github.com/NVIDIA/cccl/blob/main/CONTRIBUTING.md#cub

That should include a section on using `lit` directly to invoke tests.

Reviewer 2: Let's follow up with this later: #2935

   apt-get -y install python-pip
   pip install lit

Windows, Native Build/Test

> ===========================

Reviewer 2: Q: Are those instructions for native build/test on Windows (no linux), or for Windows and any native build/test on other OSs? Maybe to can rephrase the heading to make this more clear.

`Install Python <https://www.python.org/downloads/windows>`_.

Download `the get-pip.py bootstrap
script <https://bootstrap.pypa.io/get-pip.py>`_ and run it.

Install the LLVM Integrated Tester (``lit``) using a Visual Studio
command prompt:

.. code:: bash

   pip install lit

Step 1: Use the build scripts:
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

We provide build scripts for the most common test cases that should suit the majority of use cases.

In a Bash shell on Unix systems:

.. code:: bash

   ./ci/test_libcudacxx.sh

On Windows in ``x64 Native Tools Command Prompt``

.. code:: bash

   ./ci/windows/build_libcudacxx.ps1

> This should cover most users needs and can be adopted to work with different inputs:

Reviewer 2: ```suggestion
This should cover most users' needs and can be adapted to work with different inputs:
```

.. code:: bash

    -cxx  Selects the host compiler
    -std  Selects the C++ standard mode

>     -cuda Selects the cuda compiler

Reviewer 2: Q: isn't CUDA spelled ALLCAPS?
```suggestion
    -cuda Selects the CUDA compiler
```

    -arch Selects the cuda architectures

> Optionally one can directly invoke cmake

Reviewer 2: ```suggestion
Optionally, one can directly invoke cmake
```

> Optionally Step 2: Manually generate the Build Files:

Reviewer 2: ```suggestion
Optional Step 2: Manually generate the Build Files:
```
Applies to more places.

suite is relevant to a given PR. So rather than running all tests via ``ctest`` one can use lit to run a
subset of the test suite.

Optionally Step 4: Passing cmake options:
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

It is also possible to pass individual options to the cmake configuration. For example

.. code:: bash

   cmake --preset libcudacxx-nvrtc-cpp17 -DCMAKE_CUDA_ARCHITECTURES="86"
   cmake --build --preset libcudacxx-cpp17
   ctest --preset libcudacxx-lit-cpp17

Optionally Step 5: Build a subset of the test suite:
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

.. code:: bash

   cd build
   lit libcudacxx-cpp17/RELATIVE_PATH_TO_TEST_OR_SUBFOLDER -sv

This will build and run all tests within ``RELATIVE_PATH_TO_TEST_OR_SUBFOLDER`` which must be a valid path within the CCCL.
Note that the name of the top level folder is the same as the name of the preset.

If only building the tests is desired one can pass ``-Dexecutor="NoopExecutor()"`` to the lit invocation.
.. code:: bash

   cd build
   lit libcudacxx-cpp17/RELATIVE_PATH_TO_TEST_OR_SUBFOLDER -sv -Dexecutor="NoopExecutor()"

> Finally different standard modes can be tested by passing e.g ``--param=std=c++20``

Reviewer 2: ```suggestion
Finally, different standard modes can be tested by passing e.g ``--param=std=c++20``
```

NVRTC Build/Test:
=================

> NVRTC tests can be build and tested the same way as the other tests

Reviewer 2: ```suggestion
NVRTC tests can be built and tested the same way as the other tests
```

.. code:: bash

   cmake --preset libcudacxx-nvrtc-cpp17
   cmake --build --preset libcudacxx-cpp17
   ctest --preset libcudacxx-lit-cpp17

> If you want to run individual tests its again

Reviewer 2: Suggestion: Sounds a bit sloppy, maybe:
```suggestion
If you want to run individual tests, you can just run again:
```
